addons/mail/static/src/components/message/message.js

```diff
@@ -157,6 +157,9 @@ class Message extends Component {
      * @returns {string}
      */
     get datetime() {
+        if (!this.message.date) {
+            return;
```

Comment:
This is wrong, a string getter should not return undefined.

The proper change here is not to guard the getter, but to guard the part(s) of the template using it.